reactivesocket-core/src/main/java/io/reactivesocket/fragmentation/FrameFragmenter.java

```diff
@@ -0,0 +1,72 @@
+package io.reactivesocket.fragmentation;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.reactivesocket.Frame;
+import io.reactivesocket.FrameType;
+import io.reactivesocket.frame.FrameHeaderFlyweight;
+import reactor.core.publisher.Flux;
+
+public class FrameFragmenter {
+    private final int mtu;
+
+    public FrameFragmenter(int mtu) {
+        this.mtu = mtu;
+    }
+
+    public boolean shouldFragment(Frame frame) {
+        return isFragmentableFrame(frame.getType()) && FrameHeaderFlyweight.payloadLength(frame.content()) > mtu;
```

Comment:
should the mtu be expressed in entire frame length instead of payload length? would you normally set mtu to avoid fragmentation at lower levels? TCP? HTTP/2?